### PR TITLE
New transport interface

### PR DESCRIPTION
**Pull request by MDobak**

### CODE REVIEW

**Review by konstantinzolotarev — APPROVED** [review]

File: cmd/lair/README.md [hunk 1]
```diff
@@ -101,9 +101,9 @@ Lair supports JSON and YAML configuration files.
 ### Configuration reference
 
 - `transport` - Configuration parameters for transports mechanisms used to relay messages.
-    - `transport` (string) - Transport to use. Supported mechanism are: `libp2p` and `ssb`. If empty, the `libp2p` is
-      used.
-    - `libp2p` - Configuration parameters for the libp2p transport (Spire network).
+    - `transport` (string|[]string) - Transport to use. Supported mechanism are: `libp2p`, `ssb` and `webapi`. If empty,
```
Comment (teghnet) [inline cmd/lair/README.md:104]: I am not a fan of this kind of duality. Is there a good reason to introduce it?

File: pkg/config/transport/transport.go [hunk 2]
```diff
@@ -17,38 +17,51 @@ package transport
...
+	"github.com/chronicleprotocol/oracle-suite/pkg/transport/webapi"
+	"github.com/chronicleprotocol/oracle-suite/pkg/util/timeutil"
 )
 
-const LibP2P = "libp2p"
-const LibSSB = "ssb"
-const DefaultTransport = LibP2P
+const (
+	LibP2P = "libp2p"
+	LibSSB = "libssb"
```
Comment (teghnet) [inline pkg/config/transport/transport.go:49]: I am thinking that we should leave "ssb" as the name.

File: pkg/config/transport/transport.go [hunk 3]
```diff
@@ -17,38 +17,51 @@ package transport
...
+	WebAPI = "webapi"
```
Comment (teghnet) [inline pkg/config/transport/transport.go:50]: WebAPI is a very generic name - can we think of something else?

File: cmd/lair/README.md (hunk 1 above)
Comment (MDobak) [reply cmd/lair/README.md:104]: Backward compatibility. I don't want to break everything with this update ;)

File: pkg/config/transport/transport.go (hunk 2 above)
Comment (MDobak) [reply pkg/config/transport/transport.go:49]: I changed it by mistake

File: pkg/config/transport/transport.go (hunk 3 above)
Comment (MDobak) [reply pkg/config/transport/transport.go:50]: It is a transport that uses a web API  ¯\_(ツ)_/¯

**Review by teghnet — APPROVED** [review]

**Pull request merged**